spec/activerecord/debug_errors/ext/connection_adapters/connection_pool_spec.rb

```diff
@@ -18,13 +18,16 @@
 
   describe "#acquire_connection" do
     it "displays connection owners and other threads" do
+      ActiveRecord::Base.connection_pool.checkout_timeout = 0.001 # no need to delay test suite by waiting the whole full default timeout
+
       Thread.new { sleep 10 } # another thread
 
       expect {
         ActiveRecord::Base.connection # Ensure to acquire a connection
         Array.new(ActiveRecord::Base.connection_pool.size) do
           Thread.new do
-            ActiveRecord::Base.connection_pool.checkout(0.1)
+            ActiveRecord::Base.connection_pool.checkout
+            sleep 0.001
```

Comment:
Connection pool handling in Rails 7.1 is so smarter that the existing test cannot cause `ConnectionTimeoutError`. To cause it, I tweaked the code.

ref: 1688bd02da7c7d3cfa05d063dea9e947b953e12a

Reply:
I'm afraid that this test becomes flaky because some threads might ends before the last threads call `checkout`. How about the following changes?

```diff
diff --git a/spec/activerecord/debug_errors/ext/connection_adapters/connection_pool_spec.rb b/spec/activerecord/debug_errors/ext/connection_adapters/connection_pool_spec.rb
index 7a86bc7..6f2b1fe 100644
--- a/spec/activerecord/debug_errors/ext/connection_adapters/connection_pool_spec.rb
+++ b/spec/activerecord/debug_errors/ext/connection_adapters/connection_pool_spec.rb
@@ -20,11 +20,20 @@ RSpec.describe ActiveRecord::DebugErrors::DisplayConnectionOwners do
     it "displays connection owners and other threads" do
       Thread.new { sleep 10 } # another thread

+      mutex = Mutex.new
+      cv = ConditionVariable.new
+
       expect {
         ActiveRecord::Base.connection # Ensure to acquire a connection
         Array.new(ActiveRecord::Base.connection_pool.size) do
           Thread.new do
-            ActiveRecord::Base.connection_pool.checkout(0.1)
+            mutex.synchronize do
+              ActiveRecord::Base.connection_pool.checkout(0.1)
+              cv.wait(mutex)
+            rescue
+              cv.broadcast
+              raise
+            end
           end
         end.each(&:join)
       }.to raise_error(ActiveRecord::ConnectionTimeoutError)
```

Reply:
Exactly. And to be honest, I couldn't think of any other way to do it, so I'm glad to know the cool way you suggested it. Updated d2a4654265a72dc34b10b7f81c36876bfdf74f45